dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -427,10 +428,13 @@ public VersionTypeValidation validateTestParameterSet(Set<SourceFile> sourceFile
             // Iterate over each call, determine dependencies
             Map<String, List<String>> callsToDependencies = wdlBridge.getCallsToDependencies(tempMainDescriptor.getAbsolutePath(), mainDescName);
             toolInfoMap = mapConverterToToolInfo(callsToDockerMap, callsToDependencies);
+
             // Get import files
             namespaceToPath = wdlBridge.getImportMap(tempMainDescriptor.getAbsolutePath(), mainDescName);
-        } catch (IOException | NoSuchElementException | WdlParser.SyntaxError e) {
-            throw new CustomWebApplicationException("could not process wdl into DAG: " + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } catch (WdlParser.SyntaxError ex) {
+            throw new CustomWebApplicationException(WDLHandler.WDL_PARSE_ERROR + ex.getMessage(), HttpStatus.SC_BAD_REQUEST, ex);
```

Comment:
Sorry, I didn't provide the right level of request to log the exception. Before the throw, LOG the error, making the exception the second parameter. Something like this:

```
final String msg = WDLHandler.WDL_PARSE_ERROR + ex.getMessage();
LOG.error(msg, ex);
throw new CustomWebApplicationException(msg, HttpStatus.SC_BAD_REQUEST);
```

The reason for this if this error starts happening in production, we can go look at the logs, and see what the actual error is.

I don't think you need the 3rd parameter to CustomWebapplicationException.

Reply:
Oops, I see what you mean. I thought we wanted the stack trace to be available within the thrown exception, which is why I added the extra parameter to `CustomWebapplicationException`.

[Fixed](https://github.com/dockstore/dockstore/blob/76d246baa2dabe6b004914606056c5645a7acb12/dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java#L436).